### ISSUE TITLE
ncdf4::ncvar_def error

### ISSUE THREAD

**Issue by bue21** [ErikKusch/KrigR#86]
**Describe the bug**
Dear Erik, I was trying to run your quick start tutorial and I did not have any trouble up to the CovariateSetup command when
this error occurs:                   
Error in ncdf4::ncvar_def(name = vars[i], units = units[i], dim = list(xdim,  : 
  unused Argument (Compression = 9)

I was able to download the GMTED2010 but the process to save the DEM as netcdf failed.
The netcdf is created but the Covs_ls not.

I tried with two different version (linux and windows) of R  and I will put the sessionInfo for all of them below.
Thanks for the attention
Filippo

# Reproducing the bug
Self-contained minimal working example in R to reproduce the issue.
```r
Covs_ls <- CovariateSetup(
  Training = QuickStart_Raw,
  Target = .02,
  Dir = Dir.Covariates,
  Keep_Global = TRUE
)
```

# Screenshots
If applicable, add screenshots to help explain your problem. For example, console outputs or error messages.

# Expected behavior

# Session Information
```

> sessionInfo()
R version 4.4.1 (2024-06-14)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=de_DE.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=de_DE.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=de_DE.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Vienna
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RNetCDF_2.9-2 ncdf4_1.23    terra_1.7-83  KrigR_0.9.1  

loaded via a namespace (and not attached):
 [1] gtable_0.3.6       ggplot2_3.5.1      lattice_0.22-5     vctrs_0.6.5        tools_4.4.1        doSNOW_1.0.20      generics_0.1.3    
 [8] curl_5.2.3         gstat_2.1-2        parallel_4.4.1     tibble_3.2.1       proxy_0.4-27       spacetime_1.3-2    fansi_1.0.6       
[15] xts_0.14.1         pkgconfig_2.0.3    KernSmooth_2.23-24 lifecycle_1.0.4    farver_2.1.2       compiler_4.4.1     stringr_1.5.1     
[22] FNN_1.1.4.1        progress_1.2.3     automap_1.1-12     munsell_0.5.1      codetools_0.2-19   stars_0.6-6        snow_0.4-4        
[29] class_7.3-22       tidyr_1.3.1        pillar_1.9.0       crayon_1.5.3       classInt_0.4-10    cachem_1.1.0       viridis_0.6.5     
[36] iterators_1.0.14   abind_1.4-8        foreach_1.5.2      tidyselect_1.2.1   stringi_1.8.4      purrr_1.0.2        sf_1.0-18         
[43] dplyr_1.1.4        labeling_0.4.3     cowplot_1.1.3      fastmap_1.2.0      grid_4.4.1         colorspace_2.1-1   cli_3.6.3         
[50] magrittr_2.0.3     utf8_1.2.4         e1071_1.7-16       withr_3.0.2        prettyunits_1.2.0  scales_1.3.0       sp_2.1-4          
[57] lubridate_1.9.3    timechange_0.3.0   httr_1.4.7         gridExtra_2.3      zoo_1.8-12         hms_1.1.3          memoise_2.0.1     
[64] pbapply_1.7-2      viridisLite_0.4.2  rlang_1.1.4        Rcpp_1.0.13        glue_1.8.0         DBI_1.2.3          ecmwfr_2.0.2      
[71] rstudioapi_0.17.1  reshape_0.8.9      R6_2.5.1           plyr_1.8.9         intervals_0.15.5   units_0.8-5 


> sessionInfo()
R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=German_Germany.utf8  LC_CTYPE=German_Germany.utf8    LC_MONETARY=German_Germany.utf8 LC_NUMERIC=C                    LC_TIME=German_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RNetCDF_2.9-2 terra_1.7-83  KrigR_0.9.1  

loaded via a namespace (and not attached):
  [1] DBI_1.2.3          pbapply_1.7-2      gridExtra_2.3      httr2_1.0.5        remotes_2.5.0      rlang_1.1.3        magrittr_2.0.3     e1071_1.7-16      
  [9] compiler_4.3.3     callr_3.7.6        vctrs_0.6.5        maps_3.4.2         ecmwfr_2.0.2       profvis_0.4.0      stringr_1.5.1      pkgconfig_2.0.3   
 [17] crayon_1.5.3       fastmap_1.2.0      ellipsis_0.3.2     labeling_0.4.3     utf8_1.2.4         promises_1.3.0     sessioninfo_1.2.2  tzdb_0.4.0        
 [25] ps_1.7.6           automap_1.1-12     purrr_1.0.2        cachem_1.1.0       jsonlite_1.8.8     progress_1.2.3     later_1.3.2        reshape_0.8.9     
 [33] parallel_4.3.3     prettyunits_1.2.0  R6_2.5.1           stringi_1.8.4      pkgload_1.4.0      lubridate_1.9.3    stars_0.6-6        Rcpp_1.0.12       
 [41] iterators_1.0.14   usethis_3.0.0      zoo_1.8-12         snow_0.4-4         base64enc_0.1-3    readr_2.1.5        FNN_1.1.4.1        httpuv_1.6.15     
 [49] timechange_0.3.0   tidyselect_1.2.1   rstudioapi_0.17.1  abind_1.4-8        viridis_0.6.5      codetools_0.2-19   miniUI_0.1.1.1     processx_3.8.4    
 [57] curl_5.2.3         pkgbuild_1.4.4     lattice_0.22-5     tibble_3.2.1       intervals_0.15.5   plyr_1.8.9         shiny_1.9.1        withr_3.0.2       
 [65] askpass_1.2.1      desc_1.4.3         sf_1.0-18          urlchecker_1.0.1   units_0.8-5        proxy_0.4-27       xts_0.14.1         pillar_1.9.0      
 [73] KernSmooth_2.23-22 foreach_1.5.2      ncdf4_1.23         generics_0.1.3     sp_2.1-4           spacetime_1.3-2    IRdisplay_1.1      hms_1.1.3         
 [81] ggplot2_3.5.1      munsell_0.5.1      scales_1.3.0       xtable_1.8-4       class_7.3-22       glue_1.7.0         openeo_1.3.1       tools_4.3.3       
 [89] dendextend_1.18.1  data.table_1.15.4  fs_1.6.3           cowplot_1.1.3      grid_4.3.3         tidyr_1.3.1        crosstalk_1.2.1    devtools_2.4.5    
 [97] colorspace_2.1-1   repr_1.1.7         cli_3.6.2          rappdirs_0.3.3     gstat_2.1-2        fansi_1.0.6        viridisLite_0.4.2  dplyr_1.1.4       
[105] doSNOW_1.0.20      gtable_0.3.6       digest_0.6.35      classInt_0.4-10    htmlwidgets_1.6.4  farver_2.1.2       memoise_2.0.1      htmltools_0.5.8.1 
[113] lifecycle_1.0.4    leaflet_2.2.2      httr_1.4.7         mime_0.12          jose_1.2.1         openssl_2.2.2 
```

**Comment by ErikKusch** [comment]
Fixed on the development branch with version 0.9.2. Thanks so much for bringing this to my attention!

**Comment by bue21** [comment]
thanks a lot, it worked now!
